### PR TITLE
Add Raspberry Pi GLES context init code

### PR DESCRIPTION
**Pull request by vanfanel**
SDL/OpenGL(GLES)

Add the init code for EGL context initialization so GLES1 is used on the Raspberry Pi for final rendering.
The class OpenGLRPIGraphicsManager was added. It's located in backends/graphics/glesrpi, and is instantiated from backends/platform/sdl/sdl.cpp in OSystem_SDL::initBackend.
Raspberry Pi is autodetected at build time in the configure script.

### CODE REVIEW

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Which forbidden functions do you require?

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+static struct {
+	EGLDisplay display;
+	EGLConfig config;
+	EGLContext context;
+	EGLSurface surface;
+	uint32_t width, height, refresh;
+} eglInfo;
+
+static DISPMANX_RESOURCE_HANDLE_T   resource_bg;
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Why is it static and not part of the relevant class?

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    
+	// initialize the EGL display connection
+	EGLBoolean result = eglInitialize(eglInfo.display, NULL, NULL);
+	assert(EGL_FALSE != result);
+    
+	// get an appropriate EGL frame buffer configuration
+	EGLint num_config;
+
+	static const EGLint attribute_list[] =
+	{
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Wrong indentation

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+	    EGL_RED_SIZE, 8,
+	    EGL_GREEN_SIZE, 8,
+	    EGL_BLUE_SIZE, 8,
+		EGL_ALPHA_SIZE, 8,
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Wrong indentation

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+	    EGL_NONE
+	};
+	eglInfo.context = eglCreateContext(eglInfo.display, eglInfo.config, EGL_NO_CONTEXT, context_attributes);
+	assert(eglInfo.context != EGL_NO_CONTEXT);
+    
+	// create an EGL window surface
+	int32_t success = graphics_get_display_size(0, &eglInfo.width, &eglInfo.height);
+	assert(success >= 0);
+
+	vc_dispmanx_rect_set( &dst_rect, 0, 0, eglInfo.width, eglInfo.height);
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Wrong code formatting

File: backends/graphics/glesrpi/glesrpi.cpp [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+	// connect the context to the surface
+	result = eglMakeCurrent(eglInfo.display, eglInfo.surface, eglInfo.surface, eglInfo.context);
+	assert(EGL_FALSE != result);
+
+	eglSwapInterval (eglInfo.display, 1);
+}
+
+void OpenGLRPIGraphicsManager::deinit_egl(){
+    // Release context resources
+    eglMakeCurrent( eglInfo.display, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT );
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.cpp]: Wrong code formatting everywhere in this function.

File: backends/graphics/glesrpi/glesrpi.h [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef BACKENDS_GRAPHICS_GLESKMS_GRAPHICS_H
```
Comment (sev-) [inline backends/graphics/glesrpi/glesrpi.h]: This should be BACKENDS_GRAPHICS_GLESRPI_GRAPHICS_H

File: configure [hunk 8]
```diff
@@ -1510,6 +1510,15 @@ if test "$_release_build" = yes; then
 	DEFINES="$DEFINES -DRELEASE_BUILD"
 fi
 
+#
+# Detetect if we're running on a Raspberry Pi
+#
+if grep -q BCM2708 /proc/cpuinfo; then
+	echo "Raspberry Pi detected."
```
Comment (sev-) [inline configure]: Any chance for the cross-compilation?

File: configure [hunk 9]
```diff
@@ -4049,6 +4058,17 @@ fi
 define_in_config_if_yes "$_opengl" "USE_OPENGL"
 define_in_config_if_yes "$_opengles" "USE_GLES"
 
+if test "$_use_glesrpi" = yes ; then
+	echo "\nActivating custom-context GLES RPI..."
+	_opengles=yes
+	CXXFLAGS="$CXXFLAGS -I/opt/vc/include -I/opt/vc/include/interface/vmcs_host/linux/ -I/opt/vc/include/interface/vcos/pthreads"
```
Comment (sev-) [inline configure]: Yes, those must be autodetected.

File: configure [hunk 10]
```diff
@@ -4049,6 +4058,17 @@ fi
...
+	LIBS="$LIBS -L/opt/vc/lib -lGLESv1_CM -lEGL -lm -lbcm_host -lvcos -lvchiq_arm"
```
Comment (sev-) [inline configure]: Are there any library version requirements?

**Comment by sev-** [comment]
@vanfanel Any updates?

**Comment by sev-** [comment]
Or should it be closed in favour of #562?

**Comment by bluegr** [comment]
I've talked with Vanfanel on IRC. He said that this can be closed in favor of #562. Thus, closing it

**Pull request closed without merging**